web/libs/editor/src/tags/visual/Collapse.jsx

```diff
@@ -23,76 +24,107 @@ const { Panel } = Collapse;
+    value: types.optional(types.string, ""),
 
-  size: types.optional(types.string, "4"),
-  style: types.maybeNull(types.string),
+    children: Types.unionArray([
+      "view",
+      "header",
+      "labels",
+      "label",
+      "table",
+      "taxonomy",
+      "choices",
+      "choice",
+      "rating",
+      "ranker",
+      "rectangle",
+      "ellipse",
+      "polygon",
+      "keypoint",
+      "brush",
+      "rectanglelabels",
+      "ellipselabels",
+      "polygonlabels",
+      "keypointlabels",
+      "brushlabels",
+      "hypertextlabels",
+      "text",
+      "audio",
+      "image",
+      "hypertext",
+      "audioplus",
+      "list",
+      "dialog",
+      "textarea",
+      "pairwise",
+      "style",
+      "label",
+      "relations",
+      "filter",
+      "timeseries",
+      "timeserieslabels",
+      "paragraphs",
+      "paragraphlabels",
+    ]),
+  })
+  .views((self) => ({
+    // Indicates that it could exist without information about objects, taskData and regions
+    get isIndependent() {
+      // if value starts with $ it's related to the data but in case of Panel it's just affect title
+      // we may still want to show it even if there is no data
+      // if (self.value && self.value[0] === "$") return false;
 
-  _value: types.optional(types.string, ""),
-  value: types.optional(types.string, ""),
+      // In other cases Panel can exist independent if it hase some independent children
+      return !!self.children?.some((c) => {
+        return c.isIndependent === true;
+      });
+    },
+  }));
 
-  bordered: types.optional(types.boolean, false),
-  accordion: types.optional(types.boolean, true),
+const Model = types
+  .model({
+    type: "collapse",
 
-  children: Types.unionArray(["panel"]),
-});
+    size: types.optional(types.string, "4"),
+    style: types.maybeNull(types.string),
+
+    _value: types.optional(types.string, ""),
+    value: types.optional(types.string, ""),
+
+    bordered: types.optional(types.boolean, false),
+    accordion: types.optional(types.boolean, true),
+
+    children: Types.unionArray(["panel"]),
+  })
+  .views((self) => ({
+    get store() {
+      return getRoot(self);
+    },
+    // Indicates that it could exist without information about objects, taskData and regions
+    get isIndependent() {
+      // It is independent if some panels in it are so
+      return !!self.children?.some((c) => {
+        return c.isIndependent === true;
+      });
+    },
+  }));
 
 const CollapseModel = types.compose("CollapseModel", Model, ProcessAttrsMixin);
 
 const HtxCollapse = observer(({ item }) => {
   return (
     <Collapse bordered={item.bordered} accordion={item.accordion}>
       {item.children
-        .filter((i) => i.type === "panel")
+        .filter(
+          (i) =>
+            i.type === "panel" &&
+            ((!isFF(FF_BULK_ANNOTATION) && !item.store.hasInterface("bulk:annotation")) || i.isIndependent),
```

Comment:
can we move `(!isFF(FF_BULK_ANNOTATION) && !item.store.hasInterface("bulk:annotation")` to a variable outside the filter that describes what this means? - this way we dont need to recalculate it for every `i` when i doesn't affect it.
this will also help better define what this is making it more readable

Reply:
Now I think that the condition was also wrong.